pkg/controllers/nodeclaim/lifecycle/liveness.go

```diff
@@ -51,6 +55,30 @@ func (l *Liveness) Reconcile(ctx context.Context, nodeClaim *v1.NodeClaim) (reco
 	if ttl := registrationTTL - l.clock.Since(registered.LastTransitionTime.Time); ttl > 0 {
 		return reconcile.Result{RequeueAfter: ttl}, nil
 	}
+	nodePool := &v1.NodePool{}
+	if err := l.kubeClient.Get(ctx, types.NamespacedName{Name: nodeClaim.Labels[v1.NodePoolLabelKey]}, nodePool); err != nil {
```

Comment:
Let's do all of this after calling delete on the nodeclaim - that way we don't hang a nodeclaim delete if we have to deal with a bunch of patch conflict requeues on the nodepool status condition.